### PR TITLE
OKO: Create unique tag for each operator build

### PR DESCRIPTION
**Pull request by viroel**
This is a follow up from[1] which had to be reverted since multiple tags were generated containing the same PR SHA. The openstack-operator script that pins all bundles while building meta-operator always expects to have a single tag that match with PR SHA, so tags cannot use the entire SHA content.
This patch adds a new BUILD_TAG which is created from PR SHA (first 20 characters) + Prow build id (unique per job run).
It also provide this build tag to the pin-bundle-images script using a new var added in[2].
This PRs changes how pj-rehearsal jobs validates content build by the job. For pj-rehearsal jobs we were always using service bundles from quay.io, since content from service operator didn't change. Now, we are using the operator built in pj-rehearsal job to build openstack-operator bundle to properly test the build workflow.

[1] https://github.com/openshift/release/pull/47321
[2] https://github.com/openstack-k8s-operators/openstack-operator/pull/618

### CODE REVIEW

**Comment by viroel** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-cinder-operator-main-cinder-operator-build-deploy-kuttl pull-ci-openstack-k8s-operators-cinder-operator-main-cinder-operator-build-deploy-tempest

**Comment by viroel** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-cinder-operator-main-cinder-operator-build-deploy-kuttl pull-ci-openstack-k8s-operators-cinder-operator-main-cinder-operator-build-deploy-tempest

**Comment by viroel** [comment]
/hold 

under testing

**Comment by viroel** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-cinder-operator-main-cinder-operator-build-deploy-kuttl

**Comment by viroel** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-cinder-operator-main-cinder-operator-build-deploy-tempest

**Comment by viroel** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-cinder-operator-main-cinder-operator-build-deploy-tempest

**Comment by viroel** [comment]
Without changing any job's step, it seems that the build continues to work:
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openstack-k8s-operators_openstack-operator/618/pull-ci-openstack-k8s-operators-openstack-operator-main-openstack-operator-build-deploy-kuttl/1744738953837154304/artifacts/openstack-operator-build-deploy-kuttl/openstack-k8s-operators-build/build-log.txt

**Comment by viroel** [comment]
Based on build logs[1]:
Pushing bundle image with the new unique build-tag:
`* The resulting image will be pushed with Docker to "quay.rdoproject.org/openstack-k8s-operators/cinder-operator-bundle:c7f027a471908e4d9deb-1744681803869851648"`

While building openstack-operator, extracting proper operator bundle content:
```
+ for BUNDLE in $(hack/pin-bundle-images.sh | tr "," " ")
+ skopeo copy docker://quay.rdoproject.org/openstack-k8s-operators/cinder-operator-bundle:c7f027a471908e4d9deb-1744681803869851648 dir:tmp/bundle_extract/tmp
```
While other continue to use ref from requirements:
```
+ for BUNDLE in $(hack/pin-bundle-images.sh | tr "," " ")
+ skopeo copy docker://quay.io/openstack-k8s-operators/barbican-operator-bundle:d50607d50e9afd36e7afecc49bb035ae993d2946 dir:tmp/bundle_extract/tmp
```
Deploying[2]:
`OPENSTACK_OPERATOR_INDEX=quay.rdoproject.org/openstack-k8s-operators/openstack-operator-index:c7f027a471908e4d9deb-1744681803869851648
`
Kuttl[3]:
`+ CINDER_IMG=quay.rdoproject.org/openstack-k8s-operators/cinder-operator-index:c7f027a471908e4d9deb-1744681803869851648`
and
`+ OPENSTACK_BUNDLE_IMG=quay.rdoproject.org/openstack-k8s-operators/openstack-operator-bundle:c7f027a471908e4d9deb-1744681803869851648
`

[1] https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/47419/rehearse-47419-pull-ci-openstack-k8s-operators-cinder-operator-main-cinder-operator-build-deploy-kuttl/1744681803869851648/artifacts/cinder-operator-build-deploy-kuttl/openstack-k8s-operators-build/build-log.txt
[2] https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/47419/rehearse-47419-pull-ci-openstack-k8s-operators-cinder-operator-main-cinder-operator-build-deploy-kuttl/1744681803869851648/artifacts/cinder-operator-build-deploy-kuttl/openstack-k8s-operators-deploy/build-log.txt
[3] https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/47419/rehearse-47419-pull-ci-openstack-k8s-operators-cinder-operator-main-cinder-operator-build-deploy-kuttl/1744681803869851648/artifacts/cinder-operator-build-deploy-kuttl/openstack-k8s-operators-kuttl/build-log.txt

**Comment by viroel** [comment]
/hold

until we merge https://github.com/openstack-k8s-operators/openstack-operator/pull/618

**Comment by viroel** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-cinder-operator-main-cinder-operator-build-deploy-tempest

**Comment by viroel** [comment]
/hold

need fix on fips-check

**Comment by viroel** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-cinder-operator-main-cinder-operator-build-deploy-tempest

**Comment by viroel** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-cinder-operator-main-cinder-operator-build-deploy-tempest

File: ci-operator/step-registry/openstack-k8s-operators/build/openstack-k8s-operators-build-commands.sh
Comment (marios) [inline ci-operator/step-registry/openstack-k8s-operators/build/openstack-k8s-operators-build-commands.sh:246]: this bit seems unrelated/tidy up (though it looks correct)

**Review by marios — APPROVED** [review]
looks right per commit message info

**Comment by viroel** [comment]
/pj-rehearse ack

Lets go since jobs are passing

**Pull request merged**
